tests/load-test.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```

Comment:
This script is only used in Linux-related workflows and is recommended to be placed in the `/pkg/linux/` directory

Reply:
`run-unit-tests.sh` is also Linux-related workflows.

https://github.com/apache/pulsar-client-node/blob/dc2c3c5feb1d51e317215ae183ccf0a1b7306353/tests/run-unit-tests.sh#L27

Reply:
IMO, It's good to:
- Put `load_test.js` under `tests/linux`
- Rename `load_test.js` with `linux_load_test.js`

It's bad to put `load_test.js` under `pkg/linux`.

Reply:
Let's try to summarize these scripts:

- `./tests/run-unit-test.sh`: Used as unit test verification for running CI, suitable for Linux platforms. So, put it in the `/tests` directory.
- `./pkg/load_test.sh`: Used as a validation binary when running CI, suitable for all platforms(Require compile first). So, put it in the `./pkg/` dir.
- `./tests/load-test.sh` and `./tests/docker-load-test.sh`:Used as validation binary on multiple platforms in Linux(Require compile first).

I think `./tests/load-test.sh` and `./tests/docker-load-test.sh` is used to validate multiple platforms and are not part of unit testing.

Reply:
@BewareMyPower Can you take a look at it? We want to release this fix to 1.8.1

Reply:
Actually `load_test.js` should not be named as "xxx_test.js`. It does not "test" anything. It's just a very trivial example script that creates a `Client` object and closes it.

A script that tries to run that script in a specific environment should be treated as "test" (test if `xxx.js` can run with the Node.js client installed in a specific env)

Reply:
> Actually load_test.js should not be named as "xxx_test.js". It does not "test" anything. It's just a very trivial example script that creates a Client object and closes it.

Yes, It is to test whether the binary package is usable, not unit testing. Similarly, `./tests/load-test.sh` and `docker-load-test.sh`  also test whether the binary package can be used(Multi-platform under Linux). 

They are all part of the compilation process, so I recommend putting them in the `pkg` directory

IMO, That only unit test-related files should be placed in the `tests` directory.

Reply:
> IMO, That only unit test-related files should be placed in the tests directory.

So you mean:
- A `tests` directory should only contain **unit tests**, not other tests.
- A `pkg` directory could contain **tests**, not only packaging scripts.

What makes `pkg` directory so different that it can contain everything?

Reply:
My perspective is, `tests` directory groups all tests into one directory, including:
1. Tests for the `src/` directory.
2. Tests for the `pkg/` directory.
3. ...

If we have some shell scripts in other directories like `bin/`, the tests for them could also be put under the `tests` directory.

Actually I want to move the `load_test.js` into the `tests` directory, but I want to avoid unnecessary project structure changes.

Reply:
It's very clear that:
- `pkg/` directiory contains scripts that package the Node.js binaries in various operation systems.
- `src/` directory containes source code
- `docs/` directory contains documents
- `example/` directory contains examples
- `tests/` directory contains tests

I do not know why do you so insist on distributing tests to different directories.

Some languages tend to put tests under the same directory with the code to test, like Golang, the source code directory contains `xxx.go` and the associated test `xxx_test.go`. It seems that your logic looks like that: the tests (`load-test.sh`) for package scripts should be put under the same directory (`pkg/linux/*.sh`). However, Node.js is different with those languages. Just like Java, the tests for source code (`src/`) are put under a separated directory (`tests/`). If you changed the rule, the for users new to this project, it's not clear to know tests are distributed into how many directories..

Reply:
I don't want to play word games like "verify" and "tests", but since you mentioned this idea, I found a reference here: https://www.quora.com/What-is-the-difference-between-testing-and-verification

> Verification: Before a system is developed, there must be a design where the basic requirements for the system are well spelt out. verification is concerned with whether the system is well-engineered, error-free, and so on. Verification will help to determine whether the software is of high quality.
> Testing refers to the process of executing an application or program with the intent of detecting potential software bugs. It also has the objective of helping the developers to find out whether the software is working according to the intended purpose.

The `load-tests.sh` should be executed **after** the binaries are built and it tests whether the binaries can be executed in a given docker container.

Reply:
I realized https://github.com/apache/pulsar-client-python/blob/main/pkg/test-wheel.sh might inspire you to put scripts under the `pkg/` directory. I don't think it's a good practice.

First, in this script, `/pulsar-client-python/wheelhouse/pulsar_client-*.whl` should be the paths inside a docker container. This script cannot be executed locally. It's more like the `docker-load-test.sh` in this PR.

However, the path of `docker-load-test.sh` is not related to the Dockerfile under `pkg/linux`. Instead, the path of `/pulsar-client-node/build/Release/pulsar.node` and `/pulsar-client-node/pulsar-client-node.tar.gz` are created by `load-test.sh`, not the `pkg/linux/Dockerfile_linux_glibc`.

Reply:
Well, In short. My point is to verify `pulsar.node` does not belong to the unit test. Although it is important, it is part of the compilation.

Let's go on and stop tangling.